tools/userland-component

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python3.5
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2021 Aurelien Larcher
+#
+
+import argparse
+import os
+import re
+import sys
+import json
+
+from bass.component import Component
+from bass.makefiles import Item
+from bass.makefiles import Keywords
+from bass.makefiles import Makefile as MK
+
+# Refactoring rules
+# 000:  Use WS_* variables instead $(WS_TOP)/* 
+#       If $(WS_TOP)/make-rules is found in an include then replace with the
+#       variable $(WS_RULES). Do the same for other variables.
+def refactor000(mk):
+    for i in iter(mk.includes):
```

Comment:
Speaking of sound names: A better name of refactor000() would be refactor_use_WS_variables()

Reply:
The naming reflects the order in which refactoring functions are supposed to be called, otherwise we may miss assumptions.